model/histogram/float_histogram.go

```diff
@@ -744,9 +839,9 @@ func (h *FloatHistogram) reconcileZeroBuckets(other *FloatHistogram) float64 {
 // targetSchema prior to iterating (without mutating FloatHistogram).
 func (h *FloatHistogram) floatBucketIterator(
 	positive bool, absoluteStartValue float64, targetSchema int32,
-) *floatBucketIterator {
+) (*floatBucketIterator, error) {
 	if targetSchema > h.Schema {
-		panic(fmt.Errorf("cannot merge from schema %d to %d", h.Schema, targetSchema))
+		return nil, fmt.Errorf("cannot merge from schema %d to %d", h.Schema, targetSchema)
```

Comment:
Do we really need this change? The invariant that targetSchema must be ≤ the schema of FloatHistogram has to be enforced by the code. If this error ever happens, it is a programming error. Therefore, it is fine to panic, and we don't have to propagate the error everywhere. (As you can see in this PR, it makes the code much mode fiddly.)

Reply:
yes, this is a difficult one. 

When we parse a histogram series expression, this is a user error instead of a programmer error, but I also agree that the code is getting more complex as a result.

This is the call site:

```go
	return p.histogramsSeries(base, inc, times, func(a, b *histogram.FloatHistogram) (*histogram.FloatHistogram, error) {
		return a.Add(b)
	})
```

Before the signature change, I had

```go
	defer func() {
		if r := recover(); r != nil {
			fmt.Println("recovered:", r)
			err = fmt.Errorf("error combining histograms: %v", r)
		}
	}()
```

Alternatives I can think of:
1. Use the recover again
2. Check for schema explicitly before calling add in histogramsSeries (possibly adding a `canBeCombined` method)
3. `TryAdd` and `Add` (first one returns an error)

Reply:
I see two options. I'll comment on the call site below.

Reply:
OK, did as advised.